### ISSUE TITLE
[package] zlib/1.2.11: Build error on Windows 10 using clang-cl

### ISSUE THREAD

**Issue by StefanoLusardi** [conan-io/conan-center-index#4028]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **zlib/1.2.11**
  * Operating System+version: **Windows 10**
  * Compiler+version: **clang-cl 10.0**
  * Docker image: 
  * Conan version: **conan 1.32.1**
  * Python version: **Python 3.7.5**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
Configuration for profile default:

[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=16
build_type=Release
[options]
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)
conan install zlib/1.2.11@_/_ -e CONAN_CMAKE_GENERATOR=Ninja -e CXX=clang-cl -e CC=clang-cl -s compiler=clang -s compiler.version=10 -s **build_type=Debug** --build zlib

### Please note:
The command works perfectly with *Release* build_type.  
The bug is at [line 142](https://github.com/conan-io/conan-center-index/blob/d2739f34ec24cdf9c96d624f48d93f0710e14638/recipes/zlib/1.2.11/conanfile.py#L142) of zlib conanfile.py: the script attempts to rename 'C:\\.conan\\cc74c4245ace\\1\\lib\\**zlibstatic.lib**' but the library is named with the debug suffix 'd' (same as for MSVC compiler).  
I think the fix should be identical to [line 135](https://github.com/conan-io/conan-center-index/blob/d2739f34ec24cdf9c96d624f48d93f0710e14638/recipes/zlib/1.2.11/conanfile.py#L135): I've applied that on my machine and it is working as expected.  

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=clang
compiler.version=10
os=Windows
os_build=Windows
[options]
[build_requires]
[env]
CC=clang-cl
CONAN_CMAKE_GENERATOR=Ninja
CXX=clang-cl
zlib/1.2.11: Forced build from source
Installing package: zlib/1.2.11
Requirements
    zlib/1.2.11 from 'conan-center' - Cache
Packages
    zlib/1.2.11:38db1396108875c94ba870e5a30c6c7c1e2a4677 - Build

Installing (downloading, building) binaries...
zlib/1.2.11: Copying sources to build folder
zlib/1.2.11: Building your package in C:\.conan\3b522013fac65\1
zlib/1.2.11: Generator cmake created conanbuildinfo.cmake
zlib/1.2.11: Calling build()
zlib/1.2.11: Found Visual Studio 16
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.4.1
** Copyright (c) 2019 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'
-- The C compiler identification is Clang 10.0.0 with MSVC-like command-line
-- The CXX compiler identification is Clang 10.0.0 with MSVC-like command-line
-- Check for working C compiler: C:/Program Files/LLVM/bin/clang-cl.exe
-- Check for working C compiler: C:/Program Files/LLVM/bin/clang-cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files/LLVM/bin/clang-cl.exe
-- Check for working CXX compiler: C:/Program Files/LLVM/bin/clang-cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan CMake Wrapper
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- Conan: Compiler Clang>=8, checking major version 10
-- Conan: Checking correct version: 10
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of off64_t
-- Check size of off64_t - failed
-- Looking for fseeko
-- Looking for fseeko - not found
-- Looking for unistd.h
-- Looking for unistd.h - not found
-- Renaming
--     C:/.conan/3b522013fac65/1/source_subfolder/zconf.h
-- to 'zconf.h.included' because this file is included with zlib
-- but CMake generates it automatically in the build directory.
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DATAROOTDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR


-- Build files have been written to: C:/.conan/3b522013fac65/1/source_subfolder/_build
zlib/1.2.11: Found Visual Studio 16
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.4.1
** Copyright (c) 2019 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'
"[16/16 0.541] "Linking C static library lib\zlibstaticd.lib
zlib/1.2.11: Package '38db1396108875c94ba870e5a30c6c7c1e2a4677' built
zlib/1.2.11: Build folder C:\.conan\3b522013fac65\1
zlib/1.2.11: Generated conaninfo.txt
zlib/1.2.11: Generated conanbuildinfo.txt
zlib/1.2.11: Generating the package
zlib/1.2.11: Package folder C:\.conan\cc74c4245ace\1
zlib/1.2.11: Calling package()
ERROR: zlib/1.2.11: Error in package() method, line 171
        self._rename_libraries()
while calling '_rename_libraries', line 143
        os.rename(current_lib, os.path.join(lib_path, "zlib.lib"))
        FileNotFoundError: [WinError 2] The system cannot find the file specified: 'C:\\.conan\\cc74c4245ace\\1\\lib\\zlibstatic.lib' -> 'C:\\.conan\\cc74c4245ace\\1\\lib\\zlib.lib'
```

</details>

**Comment by madebr** [comment]
From the settings section of you `clang-cl` profile, we cannot deduce whether you're using `clang` or `clang-cl`.

```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=clang
compiler.version=10
os=Windows
os_build=Windows
```
Using `env` is unreliable.
There are some discussions at https://github.com/conan-io/conan/issues/1839 and https://github.com/conan-io/conan/issues/7868

By default, we assume the GNU style clang has been called, which is a reasonable assumption.
I think we can *detect* clang-cl if you also set `compiler.runtime` in your profile.
So, if your debug profile can be written as below, we should be able to handle `clang-cl`.
```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=clang
compiler.runtime=MDd
compiler.version=10
os=Windows
os_build=Windows
[options]
[build_requires]
[env]
CC=clang-cl
CONAN_CMAKE_GENERATOR=Ninja
CXX=clang-cl
```

@SSE4 
Is it ok to use `compiler.runtime` for `clang` to discriminate between ordinary clang and clang-cl?

**Comment by SSE4** [comment]
> @SSE4
> Is it ok to use `compiler.runtime` for `clang` to discriminate between ordinary clang and clang-cl?

yes, I think it could be used for now

**Comment by StefanoLusardi** [comment]
Thanks for your suggestion @madebr! 
I've tried the following:
```
conan install zlib/1.2.11@_/_ -pr=./clang_cl -s build_type=Debug --build zlib
```
where the profile clang_cl is exactly what you suggested in your comment.
The problem remains the same.

As a test I have replaced **clang-cl** with **clang** for both CC and CXX variables in the [env] section and the previous command works as expected. Of course in this case it's calling GNU Clang, which is not what I am trying to achieve.

```
[...]
-- The C compiler identification is Clang 10.0.0 with GNU-like command-line
-- The CXX compiler identification is Clang 10.0.0 with GNU-like command-line
-- Check for working C compiler: C:/Program Files/LLVM/bin/clang.exe
-- Check for working C compiler: C:/Program Files/LLVM/bin/clang.exe -- works
[...]
```

Can we use the `compiler.runtime` check you suggested in order to properly patch the zlib *conanfile.py* so that the debug suffix **'d'** is taken into account for clang-cl as well as MSVC?

Thanks for your help

**Comment by dmn-star** [comment]
Can you try?
...
compiler=Visual Studio
compiler.toolset=ClangCL

**Comment by StefanoLusardi** [comment]
@dmn-star 
Using the following profile
```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=Visual Studio
compiler.toolset=ClangCL
compiler.runtime=MDd
compiler.version=10
os=Windows
os_build=Windows
[options]
[build_requires]
[env]
CC=clang-cl
CONAN_CMAKE_GENERATOR=Ninja
CXX=clang-cl
```

gives me:

```
ERROR: zlib/1.2.11: Error in build() method, line 41
        self._build_zlib()
while calling '_build_zlib', line 113
        self._build_zlib_cmake()
while calling '_build_zlib_cmake', line 81
        cmake.configure(build_dir=".")
        ConanException: CMake does not support toolsets with generator "Ninja:.Please check your conan profile to either remove the toolset, or change the CMake generator.
```

If I remove the [env] section from the profile above I get the following error:

```
ERROR: zlib/1.2.11: Error in build() method, line 41
        self._build_zlib()
while calling '_build_zlib', line 113
        self._build_zlib_cmake()
while calling '_build_zlib_cmake', line 81
        cmake.configure(build_dir=".")
        ConanException: Error 1 while executing cd . && cmake -G "Visual Studio 10 2010 Win64" -DCONAN_LINK_RUNTIME="/MDd" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="Visual Studio" -DCONAN_COMPILER_VERSION="10" -DCONAN_CXX_FLAGS="/MP8" -DCONAN_C_FLAGS="/MP8" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="C:\.conan\83a5e7ba\1" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -Wno-dev -T "ClangCL" C:\.conan\8e922bc8\1
```

**Comment by dmn-star** [comment]
Maybe because compiler.version=10 is VS2010?

[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=Visual Studio
compiler.toolset=ClangCL
compiler.runtime=MDd
compiler.version=**16**
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

**Comment by StefanoLusardi** [comment]
Ouch, yes, of course it is!
But unfortunately that is still raising the same exception:

```
ERROR: zlib/1.2.11: Error in build() method, line 41
        self._build_zlib()
while calling '_build_zlib', line 113
        self._build_zlib_cmake()
while calling '_build_zlib_cmake', line 81
        cmake.configure(build_dir=".")
        ConanException: Error 1 while executing cd . && cmake -G "Visual Studio 16 2019" -A "x64" -DCONAN_LINK_RUNTIME="/MDd" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="Visual Studio" -DCONAN_COMPILER_VERSION="16" -DCONAN_CXX_FLAGS="/MP8" -DCONAN_C_FLAGS="/MP8" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="C:\.conan\294d55c\1" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -Wno-dev -T "ClangCL" C:\.conan\7f34533\1
```

**Comment by dmn-star** [comment]
Ok. I have called cmake -G "Visual Studio 16 2019" -A "x64" -DCONAN_LINK_RUNTIME="/MDd" .... on CLI for the meaningful error message

CMake Error at conanbuildinfo.cmake:569 (message):
**Incorrect 'Visual Studio'. Toolset specifies compiler as 'MSVC' but CMake detected 'Clang'**
Call Stack (most recent call first):
conanbuildinfo.cmake:142 (conan_check_compiler)
CMakeLists.txt:6 (conan_basic_setup)

 🤔

**Comment by dmn-star** [comment]
Found https://github.com/conan-io/conan/issues/5896, should be fixed in Conan 1.33

**Comment by madebr** [comment]
@StefanoLusardi

CMake detects the clang compiler, so I think your profile should look like:
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=clang
compiler.version=10
compiler.runtime=MD
build_type=Release
[options]
[build_requires]
[env]
CC=clang-cl
CONAN_CMAKE_GENERATOR=Ninja
CXX=clang-cl
```

Can you please post the complete log?

**Comment by dmn-star** [comment]
@madebr The Visual Studio/CLangCL toolset profile is the easiest way to compile the libraries with clang-cl and it should work without any changes to Conan recipes.

If you still want to use your profile, then Conan should provide a safe way to specify clang-cl (please do not decide based on user passed compiler options).

Maybe the new compiler "clang-cl" like apple-clang or a possibility to choose a frontend for clang msvc oder gnu.

If you choose the second option, then there should also be a possibility to choose a toolchain, so you can distinguish mingw clang from windows clang.

After that all conan recipes that have extra handling for VS should be adapted accordingly for clang/clang-cl. You have the choice. :-)

**Comment by madebr** [comment]
I just found out about `compiler.toolset`.
The profile should become:
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=10
compiler.runtime=MD
compiler.toolset=ClangCL
build_type=Release
[options]
[build_requires]
[env]
CC=clang-cl
CONAN_CMAKE_GENERATOR=Ninja
CXX=clang-cl
```

What does this give?

**Comment by dmn-star** [comment]
The "working" profile https://github.com/conan-io/conan-center-index/issues/4028#issuecomment-752743624
You must skip the Conan compiler check to get it working until 1.33 released.

If you don't set compiler.runtime, conan sets it to default (MD/MDd). 

compiler.runtime=MD causes that you lib/app use the multithread DLL (shared) version of the run-time library.
https://docs.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-160

**Comment by madebr** [comment]
Thanks @dmn-star , I didn't see your profile.
Does cci's zlib build/work with this profile?

**Comment by dmn-star** [comment]
cut copy paste from #4028 (comment)
[settings]
arch=x86_64
arch_build=x86_64
build_type=Debug
compiler=Visual Studio
compiler.toolset=ClangCL
compiler.runtime=MDd
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

**Comment by dmn-star** [comment]
>Does cci's zlib build/work with this profile (with current conan's master)?

Not without the workaround you need to add -DCONAN_DISABLE_CHECK_COMPILER="On" to cmake call to get it working.

**Comment by dmn-star** [comment]
1.33 is out and I can build zlib now.

>zlib/1.2.11 (test package): Running test()
Compressed size is: 21
Compressed string is: Conan Package Manager
Compressed size is: 22
Compressed string is: x┌s╬¤K╠HL╬NLOU­M╠♥RE
ZLIB VERSION: 1.2.11


### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
arch=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.toolset=ClangCL
compiler.version=16
os=Windows
[options]
[build_requires]
[env]
```


### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
C:\Users\dmn-star\Documents\GitHub\conan-center-index> conan create recipes/zlib/1.2.11 zlib/1.2.11@ -pr=conan/profiles/win --build missing
Exporting package recipe
zlib/1.2.11 exports: File 'conandata.yml' found. Exporting it...
zlib/1.2.11 exports: Copied 1 '.yml' file: conandata.yml
zlib/1.2.11 exports_sources: Copied 2 '.txt' files: CMakeLists.txt, CMakeLists_minizip.txt
zlib/1.2.11 exports_sources: Copied 3 '.patch' files: 0001-minizip.patch, 0002-gzguts-xcode12-compile-fix.patch, 0003-cmake-fix-msys2-subsystem.patch
zlib/1.2.11: A new conanfile.py version was exported
zlib/1.2.11: Folder: C:\Users\dmn-star\.conan\data\zlib\1.2.11\_\_\export
zlib/1.2.11: Package recipe modified in export, forcing source folder removal
zlib/1.2.11: Use the --keep-source, -k option to skip it
zlib/1.2.11: Exported revision: 49ff84df591e205f90604388f3b5329a
Configuration:
[settings]
arch=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.toolset=ClangCL
compiler.version=16
os=Windows
[options]
[build_requires]
[env]

zlib/1.2.11 (test package): Installing package
Requirements
    zlib/1.2.11 from 'conan-center' - Cache
Packages
    zlib/1.2.11:5a8982bbd1cce0b949ee1732653c0f392f70efe0 - Build

Installing (downloading, building) binaries...
zlib/1.2.11: Configuring sources in C:\Users\dmn-star\.conan\data\zlib\1.2.11\_\_\source
Downloading zlib-1.2.11.tar.gz completed [593.46k]

zlib/1.2.11: Copying sources to build folder
zlib/1.2.11: Building your package in C:\Users\dmn-star\.conan\data\zlib\1.2.11\_\_\build\5a8982bbd1cce0b949ee1732653c0f392f70efe0
zlib/1.2.11: Generator cmake created conanbuildinfo.cmake
zlib/1.2.11: Calling build()
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.19042.
-- The C compiler identification is Clang 10.0.0 with MSVC-like command-line
-- The CXX compiler identification is Clang 10.0.0 with MSVC-like command-line
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/Llvm/x64/bin/clang-cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/Llvm/x64/bin/clang-cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan CMake Wrapper
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- Conan: Skipping compiler check: Declared 'compiler.toolset'
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of off64_t
-- Check size of off64_t - failed
-- Looking for fseeko
-- Looking for fseeko - not found
-- Looking for unistd.h
-- Looking for unistd.h - not found
-- Renaming
--     C:/Users/dmn-star/.conan/data/zlib/1.2.11/_/_/build/5a8982bbd1cce0b949ee1732653c0f392f70efe0/source_subfolder/zconf.h
-- to 'zconf.h.included' because this file is included with zlib
-- but CMake generates it automatically in the build directory.
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DATAROOTDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CONAN_COMPILER_VERSION


-- Build files have been written to: C:/Users/dmn-star/.conan/data/zlib/1.2.11/_/_/build/5a8982bbd1cce0b949ee1732653c0f392f70efe0/source_subfolder/_build
Microsoft (R)-Build-Engine, Version 16.8.3+39993bd9d für .NET Framework
Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.

  Checking Build System
  Building Custom Rule C:/Users/dmn-star/.conan/data/zlib/1.2.11/_/_/build/5a8982bbd1cce0b949ee1732653c0f392f70efe0/source_subfolder/CMakeLists.txt
  zlibstatic.vcxproj -> C:\Users\dmn-star\.conan\data\zlib\1.2.11\_\_\build\5a8982bbd1cce0b949ee1732653c0f392f70efe0\source_subfolder\_build\lib\zlibstatic.lib
zlib/1.2.11: Package '5a8982bbd1cce0b949ee1732653c0f392f70efe0' built
zlib/1.2.11: Build folder C:\Users\dmn-star\.conan\data\zlib\1.2.11\_\_\build\5a8982bbd1cce0b949ee1732653c0f392f70efe0
zlib/1.2.11: Generated conaninfo.txt
zlib/1.2.11: Generated conanbuildinfo.txt
zlib/1.2.11: Generating the package
zlib/1.2.11: Package folder C:\Users\dmn-star\.conan\data\zlib\1.2.11\_\_\package\5a8982bbd1cce0b949ee1732653c0f392f70efe0
zlib/1.2.11: Calling package()
zlib/1.2.11 package(): Packaged 2 '.h' files: zconf.h, zlib.h
zlib/1.2.11 package(): Packaged 1 '.lib' file: zlib.lib
zlib/1.2.11 package(): Packaged 1 file: LICENSE
zlib/1.2.11: Package '5a8982bbd1cce0b949ee1732653c0f392f70efe0' created
zlib/1.2.11: Created package revision 83fcd92a92e8c6be2685a42bc0347c83
zlib/1.2.11 (test package): Generator txt created conanbuildinfo.txt
zlib/1.2.11 (test package): Generator cmake created conanbuildinfo.cmake
zlib/1.2.11 (test package): Generator pkg_config created zlib.pc
zlib/1.2.11 (test package): Generated conaninfo.txt
zlib/1.2.11 (test package): Generated graphinfo
Using lockfile: 'C:\Users\dmn-star\Documents\GitHub\conan-center-index\recipes\zlib\1.2.11\test_package\build\c7b5a947f123be405af3fbd6b593b400b6298d9e/conan.lock'
Using cached profile from lockfile
zlib/1.2.11 (test package): Calling build()
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.19042.
-- The C compiler identification is Clang 10.0.0 with MSVC-like command-line
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/Llvm/x64/bin/clang-cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Conan: called by CMake conan helper
-- Conan: Adjusting output directories
-- Conan: Using cmake targets configuration
-- Library zlib found C:/Users/dmn-star/.conan/data/zlib/1.2.11/_/_/package/5a8982bbd1cce0b949ee1732653c0f392f70efe0/lib/zlib.lib
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- This project seems to be plain C, using 'Clang' compiler
-- Conan: Skipping compiler check: Declared 'compiler.toolset'
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DATAROOTDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CONAN_COMPILER_VERSION


-- Build files have been written to: C:/Users/dmn-star/Documents/GitHub/conan-center-index/recipes/zlib/1.2.11/test_package/build/c7b5a947f123be405af3fbd6b593b400b6298d9e
Microsoft (R)-Build-Engine, Version 16.8.3+39993bd9d für .NET Framework
Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.

  Checking Build System
  Building Custom Rule C:/Users/dmn-star/Documents/GitHub/conan-center-index/recipes/zlib/1.2.11/test_package/CMakeLists.txt
C:\Users\dmn-star\Documents\GitHub\conan-center-index\recipes\zlib\1.2.11\test_package\test.c(24,41): warning : format specifies type 'unsigned long' but the argument has type 'unsigned long long' [-Wformat] [C:\Users\dmn-star\Documents\GitH 
ub\conan-center-index\recipes\zlib\1.2.11\test_package\build\c7b5a947f123be405af3fbd6b593b400b6298d9e\test_zlib.vcxproj]
C:\Users\dmn-star\Documents\GitHub\conan-center-index\recipes\zlib\1.2.11\test_package\test.c(26,41): warning : format specifies type 'unsigned long' but the argument has type 'unsigned long long' [-Wformat] [C:\Users\dmn-star\Documents\GitH 
ub\conan-center-index\recipes\zlib\1.2.11\test_package\build\c7b5a947f123be405af3fbd6b593b400b6298d9e\test_zlib.vcxproj]
  test_zlib.vcxproj -> C:\Users\dmn-star\Documents\GitHub\conan-center-index\recipes\zlib\1.2.11\test_package\build\c7b5a947f123be405af3fbd6b593b400b6298d9e\bin\test.exe
  Building Custom Rule C:/Users/dmn-star/Documents/GitHub/conan-center-index/recipes/zlib/1.2.11/test_package/CMakeLists.txt
zlib/1.2.11 (test package): Running test()
Compressed size is: 21
Compressed string is: Conan Package Manager
Compressed size is: 22
Compressed string is: x┌s╬¤K╠HL╬NLOU­M╠♥RE
ZLIB VERSION: 1.2.11
```

</details>

**Comment by AndreyMlashkin** [comment]
I am trying to create a Dockerfile with clang compiler, but the profiles above does not work there. Could you please give me any hint? https://github.com/AndreyMlashkin/windows-clang-conan

**Comment by dmn-star** [comment]
Sorry for late response, I was on holiday. If you still need the help, I could take a look at it.

**Comment by AndreyMlashkin** [comment]
Ah, ok, I wanted to use LLVM/Clang

**Comment by dmn-star** [comment]
As far as I know, you cannot use clang alone on Windows and always need either VS or MinGW in addition.

**Comment by SSE4** [comment]
was fixed in #9807
